### PR TITLE
[Faucet] Upgrade ContractKit version in Faucet

### LINKED ISSUE #2779: Web invite spouting web3 errors
### Expected Behavior

No errors after fauceting/inviting is done

### Current Behavior

https://console.firebase.google.com/project/celo-faucet/functions/logs?severity=DEBUG
```
Error: Invalid JSON RPC response: ""
    at Object.InvalidResponse (/srv/functions/node_modules/web3-core-helpers/src/errors.js:42:16)
```

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Upgrade ContractKit version in Faucet
- Add retry logic to tx sending
- Remove unused configs
- Remove dead code
- Remove duplication with code in @celo/utils
- Remove unneeded yarn.lock file
- Tighten up firebase db access rules

### Tested

In Faucet staging using local web dev build

### Other changes

Fix typo in utils log

### Related issues

- Fixes #2779 

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/faucet/src/database-helper.ts [hunk 1]
```diff
@@ -97,29 +100,35 @@ function buildHandleInvite(request: RequestRecord, snap: DataSnapshot, config: N
...
     }
-    if (!isE164Number(request.beneficiary)) {
+    if (!PhoneNumberUtils.isE164Number(request.beneficiary)) {
       throw new Error('Must send to valid E164 Number.')
     }
+    console.info(`req(${snap.key}): Creating Celo Adapter`)
     const celo = new CeloAdapter({ nodeUrl: config.nodeUrl, pk: account.pk })
+    console.info(`req(${snap.key}): New kit created`)
     const { address: tempAddress, inviteCode } = generateInviteCode()
+    console.info(`req(${snap.key}): Invite code generated`)
```
Comment (aaronmgdr) [inline packages/faucet/src/database-helper.ts:110]: where do we go to view these logs

File: packages/faucet/database-rules.bolt [hunk 2]
```diff
@@ -57,23 +89,3 @@ isAllowed(ref) {
   return auth.uid == "gYOD4GPV86OlqQh33Loc5lh0Yo02" || auth.uid == "LLnGFlilnxbXrb5pZzvjiDeUJ6l2"
```
Comment (aaronmgdr) [inline packages/faucet/database-rules.bolt:89]: i thought you had written that showing these was a security issue?  did i misread?

File: packages/faucet/scripts/cli.ts [hunk 3]
```diff
@@ -203,13 +200,10 @@ function setConfig(network: string, config: Partial<NetworkConfig & TwilioParams
...
     setIfPresent('faucet_gold_amount', config.faucetGoldAmount),
     setIfPresent('faucet_dollar_amount', config.faucetDollarAmount),
-    setIfPresent('gold_token_address', config.goldTokenAddress),
-    setIfPresent('stable_token_address', config.stableTokenAddress),
-    setIfPresent('escrow_address', config.escrowAddress),
     setIfPresent('invite_gold_amount', config.inviteGoldAmount),
     setIfPresent('invite_dollar_amount', config.inviteDollarAmount),
     setIfPresent('escrow_dollar_amount', config.escrowDollarAmount),
-    setIfPresent('expirary_seconds', config.expirarySeconds),
+    setIfPresent('expirary_seconds', config.expirySeconds),
```
Comment (aaronmgdr) [inline packages/faucet/scripts/cli.ts]: should we fix the string too? `'expirary_seconds'`

**Comment by aaronmgdr** [comment]
nice job cleaning up those files

File: packages/faucet/src/database-helper.ts (hunk 1 above)
Comment (jmrossy) [reply packages/faucet/src/database-helper.ts:110]: https://console.cloud.google.com/logs/query;query=resource.type%3D%22cloud_function%22;timeRange=PT1H;summaryFields=:true:32:beginning?authuser=0&folder=true&organizationId=true&project=celo-faucet-staging

File: packages/faucet/database-rules.bolt (hunk 2 above)
Comment (jmrossy) [reply packages/faucet/database-rules.bolt:89]: It's not ideal but not exactly an issue. This is the whitelist, and it'd be better to keep them secret, but they're out there already. In theory, there's no way for a user to impersonate these uid except by logging in with our secret creds

File: packages/faucet/scripts/cli.ts (hunk 3 above)
Comment (jmrossy) [reply packages/faucet/scripts/cli.ts]: Ah yep good catch, will fix

**Review by aaronmgdr — APPROVED** [review]
lgtm

**Pull request merged**